### PR TITLE
tests: benchmarks: multicore: Add test for S2RAM and watchdog

### PR DESCRIPTION
**Pull request by nordic-segl**
Add test that combines functional check (feeding the watchdog works) with non-functional check (current consumption).

Confirm that when watchdog is enabled:
- SOC may enter low power states like IDLE or S2RAM;
- after leaving low power state, WDT will NOT fire when feed on time.

### CODE REVIEW

File: tests/benchmarks/multicore/idle_wdt/src/main.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(idle_wdt, LOG_LEVEL_INF);
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/watchdog.h>
+
+#include <nrfs_backend_ipc_service.h>
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_wdt/src/main.c:13]: to be removed

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/multicore/idle_wdt/sysbuild.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+if("${SB_CONFIG_REMOTE_BOARD}" STREQUAL "")
+  message(FATAL_ERROR "REMOTE_BOARD must be set to a valid board name")
+endif()
+
+# Add remote project
+ExternalZephyrProject_Add(
+    APPLICATION remote
+    SOURCE_DIR ${APP_DIR}/remote
+#    SOURCE_DIR ${APP_DIR}/../../power_consumption/common/remote_sleep_forever
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_wdt/sysbuild.cmake:15]: to be removed

**Review by nordicjm — APPROVED** [review]

File: tests/benchmarks/multicore/idle_wdt/remote/boards/nrf54h20dk_nrf54h20_cpurad.overlay [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		watchdog0 = &wdt010;
+	};
+};
+
+&wdt010 {
+	status = "okay";
+};
+
+&rng_hci {
+	status = "disabled";
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_wdt/remote/boards/nrf54h20dk_nrf54h20_cpurad.overlay:18]: Same like for cpuapp. Added in hope to lower current consumption. However, no impact was observed.
Shall I remove this?

File: tests/benchmarks/multicore/idle_wdt/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	aliases {
+		watchdog0 = &wdt010;
+	};
+};
+
+&wdt010 {
+	status = "okay";
+};
+
+&rng_hci {
```
[the remark this answers is not in the document]
Comment (nordic-piks) [reply tests/benchmarks/multicore/idle_wdt/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:17]: Please remove for clear picture.

File: tests/benchmarks/multicore/idle_wdt/remote/boards/nrf54h20dk_nrf54h20_cpurad.overlay (hunk 3 above)
Comment (nordic-segl) [reply tests/benchmarks/multicore/idle_wdt/remote/boards/nrf54h20dk_nrf54h20_cpurad.overlay]: removed

**Review by nordic-piks — APPROVED** [review]

**Comment by nordic-segl** [comment]
@nordicjm : I made some changes and dismissed old approves. Please re approve / review.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**